lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb

```diff
@@ -76,6 +89,14 @@ def self.details
       end
 
       def self.available_options
+        file_exists_conflicting_options_handler = proc do |other|
+          # Fail if the conflicting options are not the ones we expect.
+          # This can only happen if the config items are updated without updating the conflicting options.
+          UI.user_error!("Unexpected conflict with option #{other.key}") unless [:skip_if_exists, :if_exists].include?(other.key)
+
+          UI.important('When both :skip_if_exists and :if_exists are set, :skip_if_exists is ignored in favor of :if_exists')
```

Comment:
1. I haven't tested this live, but doesn't fastlane already prints some error in case of `conflicting_options`? Do we really have to raise a `UI.user_error!` message ourselves, as opposed to letting fastlane do it? I didn't double-check in fastlane's implementation but I'd expect that `conflict_block` might only be useful in case you want to handle/resolve the conflict automatically or whatnot?
2. `UI.user_error!` is a raising method (as highlighted by the convention of ending with `!`). As such, `UI.important` will never be executed, because the cause will have raised and crashed beforehand.

Reply:
> UI.user_error! is a raising method (as highlighted by the convention of ending with !). As such, UI.important will never be executed, because the cause will have raised and crashed beforehand.

Correct, but `UI.user_error!` will only run if the conflicting options is "unexpected" because of the `unless` clause. That is neither `:skip_if_exists` nor `:if_exists`.

Reply:
> [looking at an example of other options using conflicting_options, I don't see them using an explicit conflict_block](https://github.com/fastlane/fastlane/blob/master/supply/lib/supply/options.rb?rgh-link-date=2023-06-19T12%3A42%3A39Z#L68-L120)

[There are others that do](https://github.com/search?q=repo%3Afastlane%2Ffastlane%20conflict_block&type=code), e.g.:

```ruby
# sigh/lib/sigh/options.rb
#
conflicting_options: [:developer_id, :development],
conflict_block: proc do |option|
  UI.user_error!("You can't enable both :#{option.key} and :adhoc")
# ...                                     
conflicting_options: [:adhoc, :development],
conflict_block: proc do |option|
  UI.user_error!("You can't enable both :#{option.key} and :developer_id")
```

Reply:
@AliSoftware 

> doesn't fastlane already prints some error in case of conflicting_options?

Yes. I haven't tried it "live" either, but this can be verified running the tests without the `conflict_block`. We'll get failures like:

```
  2) Fastlane::Actions::UploadToS3Action uploading a file with invalid parameters when the file already exists on S3 logs a message without failing when if_exists is :skip, ignoring skip_if_exists
     Failure/Error: Fastlane::FastFile.new.parse(lane).runner.execute(:test)

     FastlaneCore::Interface::FastlaneError:
       Unresolved conflict between options: 'if_exists' and 'skip_if_exists'
```

The `conflict_block` as it's currently implemented will allow the conflicting options, print a warning, and defer to `:if_exists`.

I came to that implementation from my previous mental model that lacked the knowledge of Fastlane's support for conflicting options. I like that the behavior is flexible, but maybe is best if we act more strongly and do not allow conflicting options at all? After all, that's sort of what the setting is for.

What do you think?

Reply:
> Correct, but UI.user_error! will only run if the conflicting options is "unexpected" because of the unless clause. That is neither :skip_if_exists nor :if_exists.

🤦  If only I learned to scroll horizontally before commenting… 😅 

>  is best if we act more strongly and do not allow conflicting options at all? After all, that's sort of what the setting is for.

💯 👍